### ISSUE TITLE
Source map "sources" array should be treated as URLs, not paths

### ISSUE THREAD

**Issue by nex3** [postcss/postcss#1347]
A source map's `"sources"` array is a list of URLs, but PostCSS treats them as though they're file paths. This can cause a number of problems, such as incorrectly handling URL-escaped characters, but the biggest issue that comes up in practice is the handling of absolute `file://` URLs. They should be converted into absolute file paths, but currently they are not, leading to semantic incompatibilities and weird edge-case behavior.

In the following example, we get a PostCSS [`filePosition`](http://api.postcss.org/global.html#filePosition) object whose `file` field is in URL format, despite being documented as a path:

```js
// test.js
const postcss = require('postcss');
const fs = require('fs');

const root = postcss.parse(
  fs.readFileSync('test.css', 'utf8'), {from: 'test.css'});
console.log(root.source.input.origin(1, 1));
```

```shellsession
$ echo 'a {b: c}' > test.scss
$ sass --source-map-urls=absolute test.scss test.css # Dart Sass only
$ node test.js
{ file: 'file:///home/nweiz/goog/postcss/test.scss',
  line: 1,
  column: 0 }
```

**Comment by ai** [comment]
Good feature for 8.0

**Comment by ai** [comment]
What do you think if we will add `url` to `FIlePosition`?

We may need it for the case when `sources` contain non-file URLs like `https://example.com/a.css`.

**Comment by ai** [comment]
I fixed the source map generator to use URLs cea71c8

It is a first step: file → map conversation.

**Comment by ai** [comment]
The second part finished. Now we expect source map to return URL in `consumer.originalPositionFor` ded32ce

We just need to fix tests on Windows
